Tools/wasm/wasm_build.py

```diff
@@ -443,10 +548,57 @@ def clean(self, all: bool = False):
         elif self.makefile.exists():
             self.run_make("clean")
 
+    def build_emports(self, force: bool = False):
+        """Pre-build emscripten ports"""
+        platform = self.host.platform
+        if platform.ports is None or platform.cc is None:
+            raise ValueError("Need ports and CC command")
+
+        embuilder_cmd = [os.fspath(platform.ports)]
+        embuilder_cmd.extend(self.host.embuilder_args)
+        if force:
+            embuilder_cmd.append("--force")
+
+        ports_cmd = [os.fspath(platform.cc)]
+        ports_cmd.extend(self.host.emport_args)
+        if self.target:
+            ports_cmd.extend(self.target.emport_args)
+
+        if self.dynamic_linking:
+            # trigger PIC build
+            ports_cmd.append("-sMAIN_MODULE")
+            embuilder_cmd.append("--pic")
+        if self.pthreads:
+            # trigger multi-threaded build
+            ports_cmd.append("-sUSE_PTHREADS")
+            # embuilder_cmd.append("--pthreads")
```

Comment:
Should this go? Or is this here for when it will be useful?
```suggestion
```

Reply:
It's a reminder. I have a PR for next Emscripten version that adds the option.